File: 30 Days of Code/Cpp/Day25_RunningTimeComplexity.cpp
#include <bits/stdc++.h>
using namespace std;

string Prime(int num)
{
    bool flag = true;
    
    for(int j=2; j<=sqrt(num); j++){
        if(num%j == 0){
            flag = false;
            break;
        }
    }
    
    if(num<=1)flag = false;
    
    if(flag)return "Prime\n";
    else return "Not prime\n";
    
}

int main() {
   ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int t;
    cin >> t;
    
    while(t){
        int num;
        cin >> num;
        cout << Prime(num);
        t--;
    }
    
    return 0;
}

File: 30 Days of Code/Java/Day25_RunningTimeComplexity.java
import java.io.*;
import java.util.*;
import java.lang.*;

public class Solution {
    static String Prime(int num){
        boolean flag = true;
        for(int j=2; j<=Math.sqrt(num); j++){
            if(num%j == 0)
            {
                flag = false;
                break;
            }
        }
        
        if(num <= 1)return "Not prime";
        
        if(flag)return "Prime";
        else return "Not prime";
        
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        for(int i=0; i<n; i++){
            int num = sc.nextInt();
            System.out.println(Prime(num));
        }
        
    }
}